app/src/main/java/com/android/unio/model/association/AssociationRepositoryFirestore.kt

```diff
@@ -94,6 +102,30 @@ class AssociationRepositoryFirestore @Inject constructor(private val db: Firebas
             onFailure = { exception -> onFailure(exception) })
   }
 
+  /**
+   * Fetches an [Association] object from Firestore using the provided [id].
+   *
+   * @param id [String] : The id of the [Association] to fetch.
+   * @param onSuccess [(Association) -> Unit] : The callback to call when the [Association] is
+   *   fetched.
+   * @param onFailure [(Exception) -> Unit] : The callback to call when the fetch fails.
+   */
+  override fun getAssociationWithId(
```

Comment:
Good job separating the two types of databases calls. If we just use the snapchot listener, there will be way too many Firebase requests :rocket: